ClinGen/clincoded
# Connects to #1074. Case-Control form and view pages

**Technical notes:**
1. Currently, on the Case-Control form page, the majority of the form fields are not required except the following:
* Case-Control label
* Case Cohort label
* Control Cohort label
* The **Disease(s) and Phenotype(s)** section for **Case-Cohort

**Known issues:**
1. When input fields have their `type` as `number`, our form error validation currently expects to the values to be integers. As a result, any Case-Control input fields that are expected to accept decimal numbers won't be validated. For the time being, I've implemented a temporary workaround by changing the `type` of these input fields to `text`, along with their respective handlers to alert the input `parseFloat` values that are `NaN`.
2. Curators had requested the decimal numeric values to be rounded to 1 decimal place. Given the time constraint, I had discussed this with Selina to push this request to the next release.
3. Currently, on the Dashboard page, the deletion of a Case-Control/Group entry does not appear to prevent the user from accessing the entry. Via the prior histories, (e.g. creation, modification), the user can use the embedded view links to access the entry's view page.

**Steps to test:**
1. Login and use **654321** as the PMID after proceeding to the Curation-Central page.
2. Expect to see the **Case-Control** blue bar in the _CurationPalette_ on the right.
3. Click on the **Case-Control** blue bar and expect to be taken the **Case-Control** form page.
4. Fill out the form, initially with the minimum set of required fields, and submit it. Expect the form to be successfully saved and no errors shown in the Chrome devtools console.
5. Return to the Curation-Central page and expect to see the newly created Case-Control entry in the _CurationPalette_.
6. Click on the **Edit** link of the Case-Control entry and expect to be taken back to the Case-Control form page. 
7. Add/modify other form field values and submit the form again. Expect the form to be successfully saved and no errors shown in the Chrome devtools console.
8. Return to the Curation-Central page and click on the **View** link of the Case-Control entry. Expect to be taken to the Case-Control view page.
9. Go to the Dashboard page now and expect to see the history of creating and modifying the Case-Control entry.
10. Now go back to the Case-Control form page and click on the "Delete" button at the bottom of the page. Expect the deletion goes as expected after confirming it in the modal.
11. Lastly, expect to see the history of deleting the Case-Control entry on the Dashboard page.
